### PR TITLE
Migrate from rollup to vite

### PR DESCRIPTION
**Pull request by micksi**
- minimal solution
- support tree shaking
- umd build instead of cjs and iife
- TODO: check ssr usage is still working

Since `rollup-plugin-vue` has been deprecated in favor of `@vite/plugin-vue`, and the general community is working towards vite as the standard built tool for vue libraries and application, I have migrated the library to use vite for the vue3 branch. It is a minimal solution and further configuration might be needed to satisfy all needs. It does however support tree shaking when importing single components in SFC.

### CODE REVIEW

**Comment by philefstat** [comment]
This worked great for me 👍 Thanks very much. My bundle size went down by ~80% 😅

**Comment by xeonicca** [comment]
how do I use this branch as a replacement for the time being? thank you

**Comment by micksi** [comment]
@rektdeckard can you give this PR some love 😊 To not break existing setups the package should probably be released as a 2.0 release

**Comment by dacohen** [comment]
We tried to include phosphor-vue in our Vue3/Vite project, and besides generating a massive bundle, the build process was consuming almost 4 GB of memory, causing our CI/CD to crash.

This PR appears to solve both of these problems. Is there a specific reason why this hasn't been merged yet?

**Comment by rektdeckard** [comment]
Hey @micksi, @philefstat, @dacohen and everyone else watching here. Apologies for the total disregard to your very real concern about bundle size and tree-shaking. I'll be reviewing and testing this today, hope to be able to push this live soon alongside a slew of other 2.0 updates across our other packages. Thanks for being patient for so long 🥴

File: package.json [hunk 1]
```diff
@@ -6,10 +6,10 @@
   "license": "MIT",
   "repository": "phosphor-icons/phosphor-vue",
   "homepage": "https://phosphoricons.com",
-  "main": "dist/phosphor-vue.ssr.js",
-  "browser": "dist/esm/entry.js",
-  "module": "dist/esm/entry.js",
-  "unpkg": "dist/phosphor-vue.min.js",
```
Comment (rektdeckard) [inline package.json:12]: Please keep the `unpkg` entry, and point to the UMD bundle... I think Unpkg falls back to the `browser` entry by default, but let's keep it there to be safe.

**Review by rektdeckard — CHANGES_REQUESTED** [review]
This looks great @micksi, thanks for taking the time to do this. Wish I had gotten to this long ago, but work and life gets in the way, and I am not a Vue dev.

Please add the `unpkg` entry back to `package.json` and delete `package-lock.json` (we are a `yarn` family 😆) and I'll merge this ASAP.

**Comment by rektdeckard** [comment]
By the way, We'll be giving a lot more attention to the Phosphor libraries in the near future, have a ton of cleanup and new icons in flight. Hope to push this live within the next week or two as v2.0 (vue2) and v5.0 (vue3). As I am not myself a Vue dev, and we lost our maintainer who initially started this port, we're looking for someone interested in coming on as a maintainer if anyone is interested.

**Comment by dnlsndr** [comment]
Hi there, since I've started working on the library again and there are some new icons in the pipeline, we'll soon be deploying a new major of the it. 

You can install a preview by running `npm install @phosphor-icons/vue@next`

This new version is already based on Vite 4 and takes precautions to guarantee tree-shakability (which is a big problem in the old library).

Note that it's still in alpha and is bound to change, feel free to open any issues you encounter while using the alpha library. And feel free to open any other Issues/PRs if you see something that could be done better. 
The code for the new version can be found under the `next` branch.

**Pull request closed without merging**